app/views/admin/users/index.html.haml

```diff
@@ -23,10 +23,9 @@
             %td
               = user.id
             %td
-              - if user.confirmed?
-                confirmed
-              - else
-                unconfirmed
+              - if can? :update, user
+                = check_box_tag user.id, user.id, user.confirmed?, method: :patch, url: "/admin/users/#{user.id}/toggle_confirmation?value=", class: 'switch-checkbox', data: { size: 'small', off_color: 'warning', on_text: 'Yes', off_text: 'No' }
+
```

Comment:
Instead of the parameter **value** you should use a name that reflects what you are actually doing.
And it should be sent inside the user parameters.